MaterialSkin/Controls/MaterialExpansionPanel.cs

```diff
@@ -0,0 +1,607 @@
+                    _textHeaderHeight);
+
+                using (NativeTextRenderer NativeText = new NativeTextRenderer(g))
+                {
+                    // Draw description header text 
+                    NativeText.DrawTransparentText(
+                    _descriptionHeader,
+                    SkinManager.getLogFontByType(MaterialSkinManager.fontType.Body2),
+                     SkinManager.TextDisabledOrHintColor,
+                    headerDescriptionRect.Location,
+                    headerDescriptionRect.Size,
+                    NativeTextRenderer.TextAlignFlags.Left | NativeTextRenderer.TextAlignFlags.Middle);
+                }
+            }
+
+            if (_showCollapseExpand==true)
+            {
+                using (var formButtonsPen = new Pen(_useAccentColor ? SkinManager.ColorScheme.AccentColor : SkinManager.TextDisabledOrHintColor, 2))
+                {
+                    if (_collapse)
+                    {
+                        //Draw Expand button
+                        System.Drawing.Drawing2D.GraphicsPath pth = new System.Drawing.Drawing2D.GraphicsPath();
+                        PointF TopLeft = new PointF(_expandcollapseBounds.X + 6, _expandcollapseBounds.Y + 9);
+                        PointF MidBottom = new PointF(_expandcollapseBounds.X + 12, _expandcollapseBounds.Y + 15);
+                        PointF TopRight = new PointF(_expandcollapseBounds.X + 18, _expandcollapseBounds.Y + 9);
+                        pth.AddLine(TopLeft, MidBottom);
+                        pth.AddLine(TopRight, MidBottom);
+                        g.DrawPath(formButtonsPen, pth);
+                    }
+                    else
+                    {
+                        // Draw Collapse button
+                        System.Drawing.Drawing2D.GraphicsPath pth = new System.Drawing.Drawing2D.GraphicsPath();
+                        PointF BottomLeft = new PointF(_expandcollapseBounds.X + 6, _expandcollapseBounds.Y + 15);
+                        PointF MidTop = new PointF(_expandcollapseBounds.X + 12, _expandcollapseBounds.Y + 9);
+                        PointF BottomRight = new PointF(_expandcollapseBounds.X + 18, _expandcollapseBounds.Y + 15);
+                        pth.AddLine(BottomLeft, MidTop);
+                        pth.AddLine(BottomRight, MidTop);
+                        g.DrawPath(formButtonsPen, pth);
+                    }
+                }
+            }
+
+            if (!_collapse && _showValidationButtons)
+            {
+                //Draw divider
+                g.DrawLine(new Pen(SkinManager.DividersColor, 1), new Point(0, Height - _footerHeight), new Point(Width, Height - _footerHeight));
+
+                //Validation Button
+                using (NativeTextRenderer NativeText = new NativeTextRenderer(g))
+                {
+                    Point _textLocation = new Point(_savebuttonBounds.X + 8, _savebuttonBounds.Y);
```

Comment:
Location should align the text to the right, if the user types any other value other than "save", the text is misaligned. Take a look at the Material Label to see how I do this.

Reply:
Right, _savebuttonBounds update was missing.